universal-application-tool-0.0.1/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -131,4 +138,19 @@ public Result destroy(long programId, long blockDefinitionId) {
     return redirect(routes.AdminProgramBlockPredicatesController.edit(programId, blockDefinitionId))
         .flashing("success", "Removed the visibility condition for this block.");
   }
+
+  private PredicateValue parsePredicateValue(Scalar scalar, String value) {
+    switch (scalar.toScalarType()) {
```

Comment:
suggestion: I think this needs to be the right hand side of the operator, since some operators have a left hand side that uses a different type from the right (ex: `is one of` has a single string on the left and a list of strings on the right)

Reply:
ohh... great catch!!

Reply:
I'm wondering if this is actually blocked on the discussion about SELECTION vs SELECTIONS.

Reply:
Nvm - I think I get how to do this now

Reply:
I _think_ it's right now. I manually tested the parts of this that are currently supported. :)

Reply:
Need to add tests for `PredicateValue.java`

Reply:
done